app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -36,4 +36,28 @@ def create_in_provider_queue(manager_id, params)
       MiqTask.generic_action_with_callback(task_opts, queue_opts)
     end
   end
+
+  def update_in_provider(params)
+    manager.with_provider_connection do |connection|
+      connection.api.credentials.find(manager_ref).update_attributes!(params)
+    end
+
+    # Get the record in our database
+    # TODO: This needs to be targeted refresh so it doesn't take too long
+    task_ids = EmsRefresh.queue_refresh_task(manager)
+    task_ids.each { |tid| MiqTask.wait_for_taskid(tid) }
+
+    reload!
```

Comment:
I don't think this will work.  There's no class level method `reload!`

You could do something similar to the `create_in_provider` and return the updated inventory object.